mim/commands/install.py

```diff
@@ -1,580 +1,306 @@
-                    confirm_msg = (f'install {target_pkg} from wheel, but it '
-                                   'failed. Do you want to build it from '
-                                   'source if possible?')
-                    if click.confirm(confirm_msg):
-                        install(target_pkg, find_url, timeout, is_yes,
-                                is_user_dir)
-                    else:
-                        raise RuntimeError(
-                            highlighted_error(
-                                f'Failed to install {target_pkg}.'))
-            else:
-                raise RuntimeError(highlighted_error(error))
-
-    echo_success(f'Successfully installed {target_pkg}.')
-
-
-def looks_like_path(name: str) -> bool:
-    """Checks whether the string "looks like" a path on the filesystem.
-
-    This does not check whether the target actually exists, only judge from the
-    appearance.
+    And `mim install` will **install the 'mim' extra requirements**
+    for OpenMMLab packages if needed.
 
-    Args:
-        name (str): The string to be checked.
+    \b
+    Example:
+        > mim install mmdet mmcls
+        > mim install git+https://github.com/open-mmlab/mmdetection.git
+        > mim install -r requirements.txt
+        > mim install -e <path>
+        > mim install mmdet -i <url> -f <url>
+        > mim install mmdet --extra-index-url <url> --trusted-host <hostname>
+
+    Here we list several commonly used options.
+    For more options, please refer to `pip install --help`.
     """
-    if osp.sep in name:
-        return True
-    if osp.altsep is not None and osp.altsep in name:
-        return True
-    if name.startswith('.'):
-        return True
-    return False
+    if is_yes:
+        echo_warning(
+            'The `--yes` option has been deprecated, will have no effect.')
+    exit_code = install(list(args), index_url=index_url, is_yes=is_yes)
+    if exit_code != 0:
+        exit(exit_code)
 
 
-def is_installable_dir(name: str) -> bool:
-    """Check whether path is a directory containing setup.py.
+def install(
+    install_args: List[str],
+    index_url: Optional[str] = None,
+    is_yes: bool = False,
+) -> Any:
+    """Install packages via pip and add 'mim' extra requirements for OpenMMLab
+    packages during pip install process.
 
     Args:
-        name (str): The string to be checked.
+        install_args (list): List of arguments passed to `pip install`.
+        index_url (str, optional): The pypi index url.
+        is_yes (bool, optional): Deprecated, will have no effect. Reserved for
+            interface compatibility only.
     """
-    path = osp.abspath(name)
-    if osp.isdir(path):
-        setup_py = osp.join(path, 'setup.py')
-        return osp.isfile(setup_py)
-    else:
-        return False
-
+    # add mmcv-full find links by default
+    install_args += ['-f', get_mmcv_full_find_link()]
+
+    if index_url is not None and '-i' not in install_args:
```

Comment:
Should we consider `--index-url` and `--pypi-url` ?

Reply:
Updated.
I added a warning log, prompting the user to pass parameters through index_url instead of install_args.
If multiple index_urls are passed to pip, the last one will be used.